src/dos/cdrom.h

```diff
@@ -354,18 +349,44 @@ class CDROM_Interface_Image final : public CDROM_Interface
 };
 
 #if defined (LINUX)
-class CDROM_Interface_Ioctl : public CDROM_Interface_SDL
-{
+class CDROM_Interface_Ioctl : public CDROM_Interface {
 public:
-	CDROM_Interface_Ioctl		();
-
-	bool	SetDevice		(const char* path, const int cd_number);
-	bool	GetUPC			(unsigned char& attr, char* upc);
-	bool	ReadSectors		(PhysPt buffer, const bool raw, const uint32_t sector, const uint16_t num);
-	bool	ReadSectorsHost	(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	~CDROM_Interface_Ioctl();
+
+	bool SetDevice(const char* path, const int cd_number);
+	bool GetUPC(unsigned char& attr, char* upc);
+	bool GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut);
+	bool GetAudioTrackInfo(uint8_t track, TMSF& start, unsigned char& attr);
+	bool GetAudioSub(unsigned char& attr, unsigned char& track,
+	                 unsigned char& index, TMSF& relPos, TMSF& absPos);
+	bool GetAudioStatus(bool& playing, bool& pause);
+	bool GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged,
+	                        bool& trayOpen);
+	bool ReadSectors(PhysPt buffer, const bool raw, const uint32_t sector,
+	                 const uint16_t num);
+	bool ReadSectorsHost(void* buffer, bool raw, unsigned long sector,
+	                     unsigned long num);
+	bool PlayAudioSector(const uint32_t start, uint32_t len);
+	bool PauseAudio(bool resume);
+	bool StopAudio();
+	void ChannelControl(TCtrl ctrl);
+	bool LoadUnloadMedia(bool unload);
 
 private:
-	char	device_name[512];
+	void CdAudioCallback(const uint16_t requested_frames);
+	void InitAudio(const int device_number);
+	bool IsOpen() const;
+	bool Open(const char* device_name);
+
+	int cdrom_fd                      = -1;
+	mixer_channel_t mixer_channel     = nullptr;
+	std::vector<int16_t> input_buffer     = {};
+	std::vector<AudioFrame> output_buffer = {};
+	size_t frame_position                 = 0;
+	int current_sector                = 0;
+	int sectors_remaining             = 0;
+	bool is_playing                   = false;
+	bool is_paused                    = false;
```

Comment:
Please reformat.

Reply:
It's been squashed and ran through the format script again.